### PR TITLE
[CMS] fix: remove static references to cms_domains

### LINKED ISSUE #2101: [CMS] remove references to domain from the frontend code
We should use the `supporteddomains` array returned in the `/policy` request.

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes #2101 

### Solution description

I added two helper functions, `getContractorDomains` so we can transform the `supporteddomains` array into the format we are currently working with and `getDomainName` to get the name of a domain given its type (numeric value in the BE).

I am also using the `usePolicy` hook to get access to the `supporteddomains` array.

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -486,3 +487,29 @@ export const isAnchoring = (
...
+ * Function to return the domain name given an array of contractor domains and
+ * a domain type
+ * @param {array} contractorDomains
+ * @param {number|string} op
+ */
+export const getDomainName = (contractorDomains, op) => {
+  let domainName;
+  contractorDomains.forEach((domain) => {
+    if (domain.value === op) domainName = domain.label;
+  });
```
Comment (victorgcramos) [inline src/helpers.js:513]: Liked this approach, but I would use:
```suggestion
  const { label: domainName } = contractorDomains.find(
    (domain) => domain.value === op
  );
```

File: src/components/DCC/DCC.jsx [hunk 2]
```diff
@@ -8,13 +8,15 @@ import {
...
   presentationalDccContractorType,
-  presentationalDccDomain,
   presentationalStatement,
   isRevocationDcc,
   isDccActive,
   isDccApproved
 } from "src/containers/DCC/helpers";
 import { SupportOppose, DccActions } from "src/containers/DCC/Actions";
+import { usePolicy } from "src/hooks";
+
```
Comment (amassarwi) [inline src/components/DCC/DCC.jsx:18]: not necessary

File: src/components/DCC/DCC.jsx [hunk 3]
```diff
@@ -28,12 +30,19 @@ const Dcc = ({ dcc, extended }) => {
     timereviewed
   } = dcc;
 
+  const {
+    policy: { supporteddomains }
+  } = usePolicy();
+
```
Comment (amassarwi) [inline src/components/DCC/DCC.jsx:36]: not necessary

File: src/components/DCC/DCC.jsx [hunk 4]
```diff
@@ -28,12 +30,19 @@ const Dcc = ({ dcc, extended }) => {
...
+  const contractorDomains = getContractorDomains(supporteddomains);
+
```
Comment (amassarwi) [inline src/components/DCC/DCC.jsx:38]: not necessary

File: src/components/DCC/DraftDcc.jsx [hunk 5]
```diff
@@ -4,13 +4,21 @@ import { Button, Text } from "pi-ui";
...
   presentationalStatement
 } from "src/containers/DCC/helpers";
 import styles from "./DraftDcc.module.css";
+import { usePolicy } from "src/hooks";
+import { getContractorDomains, getDomainName } from "src/helpers";
 
 const DraftDcc = ({ draft, onDelete }) => {
+  const {
+    policy: { supporteddomains }
+  } = usePolicy();
```
Comment (amassarwi) [inline src/components/DCC/DraftDcc.jsx:16]: not necessary

File: src/components/DccForm/DccForm.jsx [hunk 6]
```diff
@@ -47,6 +47,12 @@ const DccForm = React.memo(function DccForm({
   isUserValid,
   isSupervisor
 }) {
+  const {
+    policy: { supporteddomains }
+  } = usePolicy();
+
```
Comment (amassarwi) [inline src/components/DccForm/DccForm.jsx:53]: not necessary

File: src/components/DccForm/DccForm.jsx [hunk 7]
```diff
@@ -47,6 +47,12 @@ const DccForm = React.memo(function DccForm({
...
+  const contractorDomains = getContractorDomains(supporteddomains);
+
```
Comment (amassarwi) [inline src/components/DccForm/DccForm.jsx:55]: not necessary

File: src/containers/User/Detail/ManageContractor/ManageDccForm.jsx [hunk 8]
```diff
@@ -38,6 +39,12 @@ const ManageDccForm = ({ onUpdate, user }) => {
     proposalsowned = []
   } = user;
 
+  const {
+    policy: { supporteddomains }
+  } = usePolicy();
+
```
Comment (amassarwi) [inline src/containers/User/Detail/ManageContractor/ManageDccForm.jsx:45]: not necessary

File: src/containers/User/Detail/ManageContractor/ManageDccForm.jsx [hunk 9]
```diff
@@ -38,6 +39,12 @@ const ManageDccForm = ({ onUpdate, user }) => {
...
+  const contractorDomains = getContractorDomains(supporteddomains);
+
```
Comment (amassarwi) [inline src/containers/User/Detail/ManageContractor/ManageDccForm.jsx:47]: not necessary

File: src/containers/User/Detail/ManageContractor/UserView.jsx [hunk 10]
```diff
@@ -92,6 +88,12 @@ const ManageContractorUserView = ({
     proposalsByToken
   );
   const { supervisors } = useSupervisors();
+  const {
+    policy: { supporteddomains }
+  } = usePolicy();
+
```
Comment (amassarwi) [inline src/containers/User/Detail/ManageContractor/UserView.jsx:94]: not necessary

File: src/containers/User/Search/Search.jsx [hunk 11]
```diff
@@ -42,6 +45,11 @@ const UserSearch = ({ TopBanner, PageDetails, Main, Title }) => {
   const { onSearchUser, searchResult, isCMS } = useSearchUser();
   const [searchError, setSearchError] = useState(null);
   const [foundUsers, setFoundUsers] = useState([]);
+  const {
+    policy: { supporteddomains }
+  } = usePolicy();
+
```
Comment (amassarwi) [inline src/containers/User/Search/Search.jsx:51]: not necessary

**Pull request merged**